### PR TITLE
Story/gra 1225

### PR DESCRIPTION
**Pull request by afeiszli**
## Describe your changes

Added StunList and removed StunHost. StunList defaults to the server + 4 public servers (2 of ours, 2 of googles).

StunList is sent to clients (see: https://github.com/gravitl/netclient/pull/246)

## Provide Issue ticket number if applicable/not in title

GRA-1225

## Provide testing steps

1. Spin up server, use default stun list
2. add client using matching branch for client build
3. Test with a local machine and a cloud machine. Should not use proxy unless NatType is asymetric nat or double nat (can see in client logs under HOSTINFO)
4. confirm connections work as normal

## Checklist before requesting a review
- [x] My changes affect only 10 files or less.
- [x] I have performed a self-review of my code and tested it.
- [x] If it is a new feature, I have added thorough tests, my code is <= 1450 lines.
- [ ] If it is a bugfix, my code is <= 200 lines.
- [ ] My functions are <= 80 lines.
- [x] I have had my code reviewed by a peer.
- [ ] My unit tests pass locally.
- [x] Netmaker is awesome.

### CODE REVIEW

File: compose/docker-compose.ee.yml [hunk 1]
```diff
@@ -14,6 +14,7 @@ services:
       EMQX_REST_ENDPOINT: "http://mq:18083"
       SERVER_NAME: "NETMAKER_BASE_DOMAIN"
       STUN_DOMAIN: "stun.NETMAKER_BASE_DOMAIN"
```
Comment (0xdcarns) [inline compose/docker-compose.ee.yml:16]: Is stun domain still required here?

**Review by 0xdcarns — CHANGES_REQUESTED** [review]

File: models/structs.go [hunk 2]
```diff
@@ -235,7 +235,7 @@ type ServerConfig struct {
 	Broker      string `yaml:"broker"`
 	Is_EE       bool   `yaml:"isee"`
 	StunPort    int    `yaml:"stun_port"`
-	StunHost    string `yaml:"stun_host"`
+	StunList    string `yaml:"stun_list"`
```
Comment (abhishek9686) [inline models/structs.go:238]: it would be better to store stun_list  as array instead of a string

**Review by abhishek9686 — CHANGES_REQUESTED** [review]

File: models/structs.go (hunk 2 above)
Comment (afeiszli) [reply models/structs.go]: updated

File: models/structs.go [hunk 3]
```diff
@@ -261,3 +261,8 @@ type JoinData struct {
 	Node Node   `json:"node" yaml:"node"`
 	Key  string `json:"key" yaml:"key"`
 }
+
+type StunServer struct {
```
Comment (0xdcarns) [inline models/structs.go:265]: missing comment for public type

File: servercfg/serverconf.go [hunk 4]
```diff
@@ -604,3 +635,32 @@ func IsProxyEnabled() bool {
...
+
+// parseStunList - turn string into slice of StunServers
+func parseStunList(stunString string) ([]models.StunServer, error) {
+	var err error
+	stunServers := []models.StunServer{}
+	stuns := strings.Split(stunString, ",")
+	if len(stuns) == 0 {
+		return stunServers, errors.New("no stun servers provided")
+	}
+	for _, stun := range stuns {
```
Comment (0xdcarns) [inline servercfg/serverconf.go:647]: would be good to also do a trim after this line, stun := strings.Trim(stun)

**Review by 0xdcarns — CHANGES_REQUESTED** [review]

**Review by 0xdcarns — APPROVED** [review]

**Pull request merged**